libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1810,6 +1810,8 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 					", " <<
 					offset <<
 					")\n";
+			else if (dynamic_cast<ArrayType const*>(_memberAccess.annotation().type))
```

Comment:
What about structs?

Reply:
I'll add them as well now. It should be the same case.

Reply:
Fixed for structs as well. Also added test to check functionality.